### ISSUE TITLE
[GAD] Unable to setup local environment with given spec

### ISSUE THREAD

**Issue by tensorsofthewall** [microsoft/LMOps#344]
I'm attempting to setup a conda environment for training GAD, but the pinned requirements are causing different issues with different packages when i try to fix them.

For example: 
```
$ pip check                                 
latex2sympy2 1.9.1 has requirement antlr4-python3-runtime==4.7.2, but you have antlr4-python3-runtime 4.9.3.
opentelemetry-proto 1.26.0 has requirement protobuf<5.0,>=3.19, but you have protobuf 6.33.1.
```

Now, if i try to fix the antlr4-python-runtime version, other packages get broken:
```
$ uv pip install antlr4-python3-runtime==4.7.2
Using Python 3.10.12 environment at: /home/sandesh.bharadwaj/.conda/envs/gad_env
Resolved 1 package in 4ms
Uninstalled 1 package in 46ms
Installed 1 package in 52ms
 - antlr4-python3-runtime==4.9.3
 + antlr4-python3-runtime==4.7.2
$ pip check
hydra-core 1.3.2 has requirement antlr4-python3-runtime==4.9.*, but you have antlr4-python3-runtime 4.7.2.
omegaconf 2.3.0 has requirement antlr4-python3-runtime==4.9.*, but you have antlr4-python3-runtime 4.7.2.
opentelemetry-proto 1.26.0 has requirement protobuf<5.0,>=3.19, but you have protobuf 6.33.1.
```

I've tried doing this using pip as well as uv, but these version requirements persist and don't allow the training scripts to run.

How to reproduce:
1. Conda environment with python=3.10.12, torch=2.6.0, and vllm=0.8.5 + run `bash local_setup.sh` inside `gad` folder.
2. Install additional requirements (`tensordict, sentence_transformers, flash_attn==2.7.3`,etc...)

**Comment by YTianZHU** [comment]
Hi @tensorsofthewall , for the environment, I suggest you run local_docker.sh to check detailed packages combinations, and follow those combinations to setup your own environment.

For the branch to use,you can use `warmup` branch for the warmup stage, and `gad` branch for the GAD stage. Please refer to readme in this repo.

**Comment by labage** [comment]
Was the generator in the warmup branch trained with SFT or GRPO?
